request.go

```diff
@@ -200,6 +201,11 @@ func (r *Request) WithName(name string) *Request {
 	r.chain.enter("WithName()")
 	defer r.chain.leave()
 
+	if r.hasExpected {
+		r.chain.fail(r.unexpectedExpectError("WithName"))
+		return r
+	}
```

Comment:
Can we please refactor this to:

```
if !r.checkOrder("WithName") {
	return r
}
```

Why:

* this way checks wont be duplicated
* if we'll add more order checks, they will be automatically used everywhere
* it matches how we do such things in other places